src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs

```diff
@@ -112,5 +112,36 @@ public static string Convert(string value)
 
             return value.Replace(',', ';');
         }
+
+        /// <summary>
+        /// Return empty list of NuGetLogCode if all lists of NuGetLogCode are not the same. 
+        /// </summary>
+        public static IEnumerable<NuGetLogCode> GetSingleOrDefaultDistinctNuGetLogCodes(IEnumerable<IEnumerable<NuGetLogCode>> nugetLogCodes)
```

Comment:
This is very specific and should not be in `MsBuildStringUtility`. Move this to `VsSolutionRestoreService`.